iati/data.py

```diff
@@ -0,0 +1,244 @@
+                if (isinstance(value_stripped, str) and
+                        sys.version_info.major > 2):  # Python v2 treats strings as byte objects by default
+                    value_stripped_bytes = value_stripped.encode()
+                else:
+                    value_stripped_bytes = value_stripped
+
+                self.xml_tree = etree.fromstring(value_stripped_bytes)
+                self._xml_str = value_stripped
+            except etree.XMLSyntaxError:
+                msg = "The string provided to create a Dataset from is not valid XML."
+                iati.utilities.log_error(msg)
+                raise ValueError(msg)
+            except (AttributeError, TypeError, ValueError):
+                msg = "Datasets can only be ElementTrees or strings containing valid XML, using the xml_tree and xml_str attributes respectively. Actual type: {0}".format(type(value))
+                iati.utilities.log_error(msg)
+                raise TypeError(msg)
+
+    @property
+    def xml_tree(self):
+        """ElementTree: A tree representation of the Dataset.
+
+        Raises:
+            TypeError: If a value that is being assigned is not an ElementTree.
+
+        Warning:
+            Does not fully hide the lxml internal workings.
+
+        Todo:
+            Check use of ElementTree in setter.
+
+        """
+        return self._xml_tree.getroottree()
+
+    @xml_tree.setter
+    def xml_tree(self, value):
+        if isinstance(value, etree._Element):  # pylint: disable=W0212
+            self._xml_tree = value
+            self._xml_str = etree.tostring(value, pretty_print=True)
+        else:
+            msg = "If setting a Dataset with the xml_property, an ElementTree should be provided, not a {0}.".format(type(value))
+            iati.utilities.log_error(msg)
+            raise TypeError(msg)
+
+    def _raw_source_at_line(self, line_number):
+        """Return the raw value of the XML source at the specified line.
+
+        Args:
+            line_number (int): A zero-indexed line number.
+
+        Returns:
+            str: The source of the XML at the specified line.
+
+        Raises:
+            TypeError: When `line_number` is not an integer.
+            ValueError: When `line_number` is negative or more than the number of lines in the file.
+
+        """
+        if not isinstance(line_number, int) or isinstance(line_number, bool):
+            raise TypeError
+
+        if line_number < 0:
+            raise ValueError
+
+        try:
+            # this is led with an empty string since the `sourceline` attribute is 1-indexed.
+            split_lines = [''] + self.xml_str.split('\n')
+            return split_lines[line_number]
+        except IndexError:
+            raise ValueError
+
+    @property
+    def version(self):
+        """Return the version that this Dataset is specified against.
```

Comment:
For clarity, should perhaps be 'Return the IATI version that this Dataset is specified against.'

Reply:
It should be consistent with the standard documentation for a `version` parameter for functions (whatever that may be).